src/main/java/com/VetApp/PurrgentCare/services/AccountServiceImplementation.java

```diff
@@ -78,7 +79,15 @@ public AccountResponse associatePeople(AssociatePeopleWithAccountRequest associa
         var people = personRepository.findAllById(personIds);
         Account account = accountRepository.findById(accountId)
                 .orElseThrow(() -> new EntityNotFoundException(String.valueOf(accountId)));
-        account.setAccountHolders(people);
+        //Had to add this loop to set the parent relationship of each child; check for duplicate children; add any new children to the parent.
+        for (Person person : people) {
+            person.setAccount(account);
+            if (!account.getAccountHolders().contains(person)) {
+                account.getAccountHolders().add(person);
+            }
+        }
+        //TODO can delete this if everything is good
+        //account.setAccountHolders(people);
```

Comment:
Just setting the accountholders was causing an issue with the hibernate cascade.
Had to set the parent for each child and ignore duplicates or the add method would still add them.

Reply:
Should put the setAccount inside the if statement. No reason to set it for duplicates.

Reply:
Does the conditional if statement, remove existing people that might have been linked to this account?
i.e. 
Account 1 had Person 1, 2 but new request changes account 1 to have Person 2, 3... Does your method have the expected result of Account 1 with Person 2, 3?

Reply:
The if statement is to keep from adding duplicate persons to the accountHolders list.
Without it Account 1 with Person 1, 2 -> Request add persons 2, 3 was ending up with Persons 1, 2, 2, 3
With it Account 1 with Person 1, 2 -> Request add persons 2, 3 results in Persons 1, 2, 3

Do we want to clear the list of accountHolders when this endpoint is hit?
My reading of the feature request suggests we should be removing the persons added from their previous accounts, not clearing the account of all people and adding a new list.

Reply:
I don't think we can have it work that way since the relationship is currently one Account to Many People.
So Person 3 could never be on more than once account.

Reply:
Fair point. Sounds like a future feature 😄